DSA-Campaign-Uplift-Estimation/src/main/webapp/Home/script-home.js

```diff
@@ -10,4 +10,143 @@
+    DSACampaignsList.innerHTML = '<p>First select a keyword campaign.</p>';
+    var keywordCampaignId = document.getElementById("keyword-campaigns").value;
+
+    if (keywordCampaignId != 0) {
+        DSACampaignsList.innerHTML = '';
+
+        fetch('/DSA-campaigns?keywordCampaignId=' + keywordCampaignId).then(response => response.json()).then(DSACampaigns => {
+
+            for(var i = firstChart; i < secondChart; i++ ){
+                drawNewBarGraph(DSACampaigns[i], secondChart-i);
+            }
+            if (firstChart == 0){
+                makePagination((DSACampaigns.length / 2) + (DSACampaigns.length % 2), (Math.log(firstChart + 1)/Math.log(2)));
+            }
+            else{
+                makePagination((DSACampaigns.length / 2) + (DSACampaigns.length % 2), (Math.log(firstChart)/Math.log(2)));
+            }            
+        });
+    }
+}
+
+// Creates pagination html for the user to see other charts by taking the existing number of pages
+// and the active page the user is on.
+function makePagination(numberOfPages,activeNumber) {
+    const pagination = document.getElementById("pagination");
+    var paginationString = "";
+    paginationString += "<input type=\"button\" id=\"previous\" onclick=\"previousPage()\" value=\"previous\" />";
+    paginationString += "<a id=\"activeNumber\">  " + (activeNumber + 1)  +"  </a>"; 
+    paginationString += "<input type=\"button\" id=\"next\" onclick=\"nextPage(" + numberOfPages + ")\" value=\"next\" />";
+    pagination.innerHTML = paginationString;
+}
+
+// Increases the page forward by increasing the active page by one and letting the display
+// charts increase to the next two and rerunning getDSACampaigns to retrieve the charts.
+function nextPage(numberOfPages) {
+    const activePage = document.getElementById("activeNumber");
+    var increasePage = parseInt(activePage.innerText);
+    
+    // The if statement checks if the user clicks nextpage but there is
+    // no more charts to look at using numberOfPages.
+    if(increasePage + 1 <= numberOfPages){
+        increasePage++;  
+        secondChart += 2;
+        firstChart += 2;  
+    }
+    activePage.innerText = increasePage;
+    getDSACampaigns();
+}
+
+// Decreases the page backward by decreasing the active page by one and letting the display
+// charts decrease to the previous two and rerunning getDSACampaigns to retrieve the charts.
+function previousPage() {
+    const activePage = document.getElementById("activeNumber");
+    var decreasePage = parseInt(activePage.innerText);
+
+    // The if statement checks if the user clicks previousPage but there is
+    // no more charts to look at using 0 because there are no negative pages.
+    if(decreasePage - 1 > 0){
+        decreasePage--;  
+        secondChart -= 2;
+        firstChart -= 2;  
+    }
+    activePage.innerText = decreasePage;
+    getDSACampaigns();
+}
+
+function drawNewBarGraph(DSACampaign, chartNumber) {
```

Comment:
change to drawBarGraphFromDSACampaign or drawDSACampaignBarGraph?

Reply:
Done.